src/datepicker/datepicker-input.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  Directive,
+  Input,
+  ComponentRef,
+  ElementRef,
+  ViewContainerRef,
+  Renderer,
+  ComponentFactoryResolver,
+  NgZone,
+  TemplateRef,
+  forwardRef
+} from '@angular/core';
+import {ControlValueAccessor, NG_VALUE_ACCESSOR} from '@angular/forms';
+
+import {NgbDate} from './ngb-date';
+import {NgbDatepicker} from './datepicker';
+import {DayTemplateContext} from './datepicker-day-template-context';
+import {NgbDateParserFormatter} from './ngb-date-parser-formatter';
+
+import {positionElements} from '../util/positioning';
+
+const NGB_DATEPICKER_VALUE_ACCESSOR = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => NgbInputDatepicker),
+  multi: true
+};
+
+@Directive({
+  selector: 'input[ngbDatepicker]',
+  exportAs: 'ngbDatepicker',
+  host: {'(change)': 'manualDateChange($event.target.value)', '(keyup.esc)': 'close()'},
+  providers: [NGB_DATEPICKER_VALUE_ACCESSOR]
+})
+export class NgbInputDatepicker implements ControlValueAccessor {
+  private _cRef: ComponentRef<NgbDatepicker> = null;
+  private _model: NgbDate;
+  private _zoneSubscription: any;
+
+  /**
+   * Reference for the custom template for the day display
+   */
+  @Input() dayTemplate: TemplateRef<DayTemplateContext>;
+
+  /**
+   * First day of the week, 0=Sun, 1=Mon, etc.
+   */
+  @Input() firstDayOfWeek: number;
+
+  /**
+   * Callback to mark a given date as disabled
+   */
+  @Input() markDisabled: (date: {year: number, month: number, day: number}) => boolean;
```

Comment:
Guys, this is not part of this PR, it is moved over from the NgbDatePicker. If you want to tackle this let's do it in a separate issue - this PR is a big one already. My quick take on it:
* yes, let's create an interface
* no, it is not the same as `NgbDate` - `NgbDate` is meant to be internal and shouldn't be exposed to the end users